### DISCUSSION TITLE
How is decided the color of the compute_kind tag ?

### DISCUSSION THREAD

**Discussion by Aluriak** [dagster-io/dagster#18253] [Q&A] [answered]
In the dagster UI, setting an asset `compute_tag` will show under the asset in lineage view a little colored box with the text inside. My first thougth was that the color is determined by hashing the text and map the hash space with colors.

However, some close texts get the same color, such as `🔐` and friends. (i didn't run a full statistical analysis, but it *feel* non random)

So, out of pure curiosity, how is the color decided ? where is the part of the source code handling this ? ( i lookupped the js code, but i don't have any knowledge of js, so i'm not sure where and what i'm looking for)

Thank you !

**Comment by TyShkan** [comment, marked as answer]
It's already discussed [here](https://github.com/dagster-io/dagster/issues/17570)

**Reply by Aluriak** [reply]
Indeed, thank you !
